shared/update_check_settings_file.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include <AppConfig.h>
+#include <juce_core/juce_core.h>
+#include <helper/version.hpp>
+
+class UpdateCheckerSettingsFile {
+public:
+    UpdateCheckerSettingsFile();
+    ~UpdateCheckerSettingsFile();
+
+    bool getAutoCheckEnabled();
+    bool setAutoCheckEnabled(bool enabled);
```

Comment:
Whilst I agree, this has quite deep changes throughout the code (from UpdateCheckerSettingsFile up through UpdateChecker and then up in to extension menu code and setup application code). We would also have to support 2 different data types in the settings file since this is stored as a bool in previous versions. Feels like a lot of additional changes for something we might never use. I think best tackle it only if we need it (not doing it now won't make that any more difficult).

Reply:
Actually, the settings file wouldn't be a problem as we could just split it out to the appropriate attribute in the XML to maintain compatibility. This is how the config is stored;
```
<?xml version="1.0" encoding="UTF-8"?>

<UpdateCheck>
  <LastReportedVersion VersionMajor="0" VersionMinor="0" VersionRevision="0"/>
  <AutoCheck OnStartUp="1"/>
</UpdateCheck>
```

Reply:
Ah, no worries if it's a big change, I'd not followed it that far down.